app/controllers/pages_controller.rb

```diff
@@ -3,14 +3,9 @@ class PagesController < ApplicationController
   helper_method :sort_key, :sort_dir
 
   def show
-    @page = Page.find(params[:id])
-    surveys = Survey.where(full_path: @page.base_path)
-    @survey_count = surveys.count
-    @answers_by_question = surveys.each_with_object(Hash.new { |h, k| h[k] = [] }) do |survey, grouped_answers|
-      survey.survey_answers.each do |survey_answer|
-        grouped_answers[survey_answer.question] << survey_answer
-      end
-    end
+    base_path = request.path.sub("/page", "")
+    @page = Page.find_by_base_path(base_path)
```

Comment:
Not sure if it's going to be confusing having base paths as part of the URL, but I can see why you've done this in terms of making URLs human readable and also making it easier to get to the details page if you know the base path.

Reply:
Cool, noted